src/core/CONVENTIONS.md

```diff
@@ -210,3 +210,8 @@ export class Plugin {
   }
 }
 ```
+
+### Naming conventions
+
+Export start and setup contracts as `IMyPluginStart` and `IMyPluginSetup`. 
+This avoids naming clashes, if everyone exported them simply as `Start` and `Setup`.
```

Comment:
The `MyPluginStart` vs `Start` is obviously correct.

However (and I know this has been discussed on slack, but) I think in core we only use the `I` prefix to dissociate 'interface types' from concrete *classes*. Most of our 'interfaces' does not use this prefix (`CoreSetup/Start` for starter)

WDYT @joshdover

Reply:
I think because we want this to be a lint rule we'll have to put it on all interfaces.  I filed https://github.com/elastic/kibana/issues/51674 to track that.  But happy to hear Josh weigh in too!

Reply:
Pierre is correct that this is how we've been applying this rule in Core.

I think that this distinction would be hard to codify well into a lint rule and breaks down when switching between `type` and `interface`.

It think this particular area points to one of the flaws with the `I`-prefix approach: it discriminates between `interface` and everything else which is not generally as useful as discriminating between concrete JavaScript structures (class, enum, function) and pure TypeScript constructs (type, interface).

I don't want to restart the entire conversation, but I wanted to explain that I don't think there's a perfect solution here if we stay with the `I`-prefix. 

That said, while I personally prefer doing it the way we've been doing it in Core because I find it more readable, it is a more complex rule. I think complexity will inevitably lead to inconsistencies. Going all-in on `I`-prefix in all cases seems like the most practical decision.

Reply:
that's the eternal `IServiceFactoryProvider` vs `ServiceFactoryProviderImpl` debate we've had in Java for 20 years now (example names are probably too short for Java though). 

Just to be sure: is this part of a bigger plan to have **all** of our ts types/interface being `I`-prefixed in the whole Kibana codebase, or are we only speaking of plugin contracts?